### PR TITLE
fix: workaround for quoted 'on' string

### PR DESCRIPTION
**Pull request by jimsheldon**

### CODE REVIEW

File: convert/harness/downgrader/yaml/parse.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ func Parse(r io.Reader) ([]*v1.Pipeline, error) {
 	docs := bytes.Split(b, []byte("\n---\n"))
 
 	for _, doc := range docs {
+		// workaround for this issue
+		// https://stackoverflow.com/questions/70849190/golang-how-to-avoid-double-quoted-on-key-on-when-marshaling-struct-to-yaml
+		doc = bytes.ReplaceAll(doc, []byte(` "on":`), []byte(` on:`))
```
Comment (eoinmcafee00) [inline convert/harness/downgrader/yaml/parse.go:27]: @bradrydzewski - from what I read in the docs of v3 https://github.com/go-yaml/yaml/tree/v3#compatibility this issue is resolved. I assume we'd break other stuff by switching to it?

Comment (jimsheldon) [reply convert/harness/downgrader/yaml/parse.go:27]: @eoinmcafee00 but we are already on the most recent version of that package? https://github.com/drone/go-convert/blob/master/go.mod#L16

Comment (eoinmcafee00) [reply convert/harness/downgrader/yaml/parse.go:27]: but where we do the yaml parsing in harness, it's a different library:
 https://github.com/drone/go-convert/blob/master/convert/harness/yaml/parse.go#L24

Comment (bradrydzewski) [reply convert/harness/downgrader/yaml/parse.go:27]: we use [ghoss/yaml](https://github.com/ghodss/yaml) to parse the yaml which depends on yaml v2.  unfortunately it is not something we can change at this time.  with that being said, the `on` keyword was removed from the spec and changed from this:

```
on:
  failure:
```

to this:

```
faiure:
  action:
```

so updating to the latest version of drone/spec would eliminate the need for this pull request

**Pull request closed without merging**